src/main/java/org/influxdb/dto/BoundParameterQuery.java

```diff
@@ -0,0 +1,125 @@
+package org.influxdb.dto;
+
+import com.squareup.moshi.JsonWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import okio.Buffer;
+
+public final class BoundParameterQuery extends Query {
+
+  private final Map<String, Object> params = new HashMap<>();
+
+  private BoundParameterQuery(final String command, final String database) {
+    super(command, database, true);
+  }
+
+  public String getParameterJsonWithUrlEncoded() {
+    try {
+      List<String> placeholders = parsePlaceHolders(getCommand());
+      assurePlaceholdersAreBound(placeholders, params);
+      String jsonParameterObject = createJsonObject(params);
+      String urlEncodedJsonParameterObject = encode(jsonParameterObject);
+      return urlEncodedJsonParameterObject;
+    } catch (IOException e) {
+      throw new RuntimeException("Couldn't create parameter JSON object", e);
+    }
+  }
+
+  private void assurePlaceholdersAreBound(final List<String> placeholders, final Map<String, Object> params) {
+    if (placeholders.size() != params.size()) {
+      throw new RuntimeException("Unbalanced amount of placeholders and parameters");
+    }
+
+    for (String placeholder : placeholders) {
+      if (params.get(placeholder) == null) {
+        throw new RuntimeException("Placeholder $" + placeholder + " is not bound");
```

Comment:
Personally I don't like to throw `RuntimeException`s (check why here https://wiki.sei.cmu.edu/confluence/display/java/ERR07-J.+Do+not+throw+RuntimeException%2C+Exception%2C+or+Throwable ).

Also, this exception will be thrown only when the query is about to be executed instead of the moment when I'm executing the `.create()` from the Builder. Would be possible to move it to the builder?

Another question: what about throwing `IllegalStateException` from the `.create()` method?

Reply:
Relevant code has been removed